test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs

```diff
@@ -209,5 +260,87 @@ internal static void GetDirectoryPaths (out string rootSourceDirectory, out stri
 			rootSourceDirectory = Path.GetFullPath (Path.Combine (artifactsBinDir, "..", "..", "test", "Mono.Linker.Tests.Cases"));
 			testAssemblyPath = Path.GetFullPath (Path.Combine (artifactsBinDir, "ILLink.RoslynAnalyzer.Tests", configDirectoryName, tfm));
 		}
+
+		internal static Task VerifyCodeFix<TAnalyzer, TCodeFix> (
+			string source,
+			string fixedSource,
+			DiagnosticResult[] baselineExpected,
+			DiagnosticResult[] fixedExpected,
+			int? numberOfIterations = null)
+			where TAnalyzer : DiagnosticAnalyzer, new()
+			where TCodeFix : CodeFixProvider, new()
+		{
+			string analyzerAttribute = GetAttributeDefinition<TAnalyzer> ();
+			string codeFixAttribute = GetAttributeDefinition<TCodeFix> ();
+			var attributeDefinitions = analyzerAttribute;
+			if (!string.Equals (analyzerAttribute, codeFixAttribute)) {
+				attributeDefinitions += codeFixAttribute;
+			}
+
+			var test = new CSharpCodeFixVerifier<TAnalyzer, TCodeFix>.Test {
+				TestCode = source + attributeDefinitions,
+				FixedCode = fixedSource + attributeDefinitions,
+			};
+			test.ExpectedDiagnostics.AddRange (baselineExpected);
+			var analyzerMSBuildPropertyOption = GetAnalyzerMSBuildPropertyOption<TAnalyzer> ();
+			test.TestState.AnalyzerConfigFiles.Add (
+						("/.editorconfig", SourceText.From (@$"
+is_global = true
+build_property.{analyzerMSBuildPropertyOption} = true"
+)));
+			if (numberOfIterations != null) {
+				test.NumberOfIncrementalIterations = numberOfIterations;
+				test.NumberOfFixAllIterations = numberOfIterations;
+			}
+			test.FixedState.ExpectedDiagnostics.AddRange (fixedExpected);
+			return test.RunAsync ();
+		}
+
+		internal static Task VerifyDiagnostic<TAnalyzer> (
+			string source,
+			DiagnosticResult[] baselineExpected)
+			where TAnalyzer : DiagnosticAnalyzer, new() =>
+			CSharpAnalyzerVerifier<TAnalyzer>.VerifyAnalyzerAsync (
+				source,
+				TestCaseUtils.UseMSBuildProperties (GetAnalyzerMSBuildPropertyOption<TAnalyzer> ()),
+				baselineExpected);
+
+		internal static DiagnosticResult GetDiagnosticResult<TAnalyzer> ()
+			where TAnalyzer : DiagnosticAnalyzer, new()
+		{
+			return CSharpAnalyzerVerifier<TAnalyzer>.Diagnostic (GetDiagnosticId<TAnalyzer> ());
+		}
+
+		private static string GetDiagnosticId<TAnalyzer> () where TAnalyzer : DiagnosticAnalyzer
```

Comment:
Looking more closely, this method's design seems busted. If the analyzer declares only one diagnostic descriptor then we can use the `CSharpAnalyzerVerifier<TAnalyzer>.Diagnostic()` method. The point of the method taking an ID is that we don't know which ID we're testing for. Below we hardcode `IL3002`, but what about `IL3000` and `IL3001`?